src/components/CurrencyInputPanel/index.tsx

```diff
@@ -90,57 +91,68 @@ export default function CurrencyInputPanel({
-            ) : null}
-            {pair ? (
-              <Text id="pair" bold>
-                {pair?.token0.symbol}:{pair?.token1.symbol}
-              </Text>
-            ) : (
-              <Text id="pair" bold>
-                {(currency && currency.symbol && currency.symbol.length > 20
-                  ? `${currency.symbol.slice(0, 4)}...${currency.symbol.slice(
-                      currency.symbol.length - 5,
-                      currency.symbol.length,
-                    )}`
-                  : currency?.symbol) || t('Select a currency')}
-              </Text>
-            )}
-            {!disableCurrencySelect && <ChevronDownIcon />}
-          </Flex>
-        </CurrencySelectButton>
-        <Flex style={{ gap: '8px' }}>
-          {token && tokenAddress && library?.provider?.isMetaMask && (
-            <MetamaskIcon
-              style={{ cursor: 'pointer' }}
-              width="16px"
-              ml="6px"
-              onClick={() => registerToken(tokenAddress, token.symbol, token.decimals)}
-            />
-          )}
-          {account && (
-            <Text onClick={onMax} color="textSubtle" fontSize="14px" style={{ display: 'inline', cursor: 'pointer' }}>
-              {!hideBalance && !!currency
-                ? t('Balance: %balance%', { balance: selectedCurrencyBalance?.toSignificant(6) ?? t('Loading') })
-                : ' -'}
-            </Text>
-          )}
+        <Flex>
+          <CurrencySelectButton
+            className="open-currency-select-button"
+            selected={!!currency}
+            onClick={() => {
+              if (!disableCurrencySelect) {
+                onPresentCurrencyModal()
+              }
+            }}
+          >
+            <Flex alignItems="center" justifyContent="space-between">
+              {pair ? (
+                <DoubleCurrencyLogo currency0={pair.token0} currency1={pair.token1} size={16} margin />
+              ) : currency ? (
+                <CurrencyLogo currency={currency} size="24px" style={{ marginRight: '8px' }} />
+              ) : null}
+              {pair ? (
+                <Text id="pair" bold>
+                  {pair?.token0.symbol}:{pair?.token1.symbol}
+                </Text>
+              ) : (
+                <Text id="pair" bold>
+                  {(currency && currency.symbol && currency.symbol.length > 20
+                    ? `${currency.symbol.slice(0, 4)}...${currency.symbol.slice(
+                        currency.symbol.length - 5,
+                        currency.symbol.length,
+                      )}`
+                    : currency?.symbol) || t('Select a currency')}
+                </Text>
+              )}
+              {!disableCurrencySelect && <ChevronDownIcon />}
+            </Flex>
+          </CurrencySelectButton>
+          {token && tokenAddress ? (
+            <Flex style={{ gap: '4px' }} alignItems="center">
+              <CopyButton
```

Comment:
Could you make it button color to `textSubtle`?

Reply:
Done